src/Test/Monad/Writer.hs

```diff
@@ -6,24 +6,40 @@ import Data.Functor (($>))
 import Control.Monad.Writer
 import Test.QuickCheck.HigherOrder (Equation(..))
 
+-- | Telling two values sequentially is equivalent to monoidally appending the two values then telling once.
+-- @
+-- 'tell' w1 '>>' 'tell' w2 = 'tell' (w1 '<>' w2)
+-- @
 tell_tell
   :: forall m w
   .  MonadWriter w m
   => w -> w -> Equation (m ())
 tell_tell w1 w2 = (tell w1 >> tell w2) :=: tell (w1 <> w2)
 
+-- | telling a 'mempty' value is equivalent to calling ('return' ()).
+-- @
+-- 'tell' 'mempty' == 'return' ()
+-- @
 tell_mempty
   :: forall m w
   .  MonadWriter w m
   => Equation (m ())
 tell_mempty = tell mempty :=: return ()
 
+-- | ('listen' . 'return') lifts a value into a monad and tuples it with 'mempty'.
+-- @
+-- 'listen' ('return' a) = 'return' (a, 'mempty')
+-- @
 listen_return
   :: forall m a w
   .  MonadWriter w m
   => a -> Equation (m (a, w))
 listen_return a = listen (return a) :=: return (a, mempty)
 
+-- |
+-- @
+-- 'listen' (m '>>=' k) = 'listen' m '>>=' \(a, wa) -> 'listen' (k a) >>= \(b, wb) -> 'return' (b, wa '<>' wb)
```

Comment:
Double check whether the backslaskes are preserved by haddock.
This line is also a bit long.

```suggestion
-- 'listen' (m '>>=' k)
-- = 'listen' m '>>=' \\(a, wa) ->
--   'listen' (k a) '>>=' \\(b, wb) ->
--   'return' (b, wa '<>' wb)
```